File: skills/clawhub/real-browser-qa-ceki-main/examples/account-login.ts
import { Browser, HumanActionDeclined, HumanActionTimeout } from '@ceki/sdk';

const br = new Browser({ token: 'YOUR_TOKEN' });
await br.connect();

const s = await br.openSession({
  mode: 'persona',
  domainHints: ['app.example.com'],
});

await s.navigate('https://app.example.com/login');

await s.injectCredentials('secret-abc-123', {
  username_selector: '#email',
  password_selector: '#password',
  submit_selector: '#login-btn',
});

try {
  const result = await s.requestHumanAction(
    '2fa',
    'Please enter the 2FA code from your authenticator app',
    120,
  );
  console.log(`2FA completed: ${result.status}`);
} catch (e) {
  if (e instanceof HumanActionDeclined) {
    console.log('Provider declined the 2FA request');
  } else if (e instanceof HumanActionTimeout) {
    console.log('Provider did not respond in time');
  } else {
    throw e;
  }
}

const page = await s.query('h1');
console.log(`Logged in. Page title: ${page.elements[0]?.textContent}`);

await s.close();
await br.close();

File: skills/clawhub/real-browser-qa-ceki-main/examples/account-login.py
"""Login to an account using credential vault + 2FA human action."""
import asyncio

from ceki_browser import Browser, HumanActionDeclined, HumanActionTimeout


async def main():
    async with Browser(token="YOUR_TOKEN") as br:
        async with await br.session(
            mode="persona",
            domain_hints=["app.example.com"],
        ) as s:
            await s.navigate("https://app.example.com/login")

            await s.inject_credentials(
                secret_id="secret-abc-123",
                target={
                    "username_selector": "#email",
                    "password_selector": "#password",
                    "submit_selector": "#login-btn",
                },
            )

            try:
                result = await s.request_human_action(
                    action_type="2fa",
                    message="Please enter the 2FA code from your authenticator app",
                    timeout_sec=120,
                )
                print(f"2FA completed: {result.status}")
            except HumanActionDeclined:
                print("Provider declined the 2FA request")
            except HumanActionTimeout:
                print("Provider did not respond in time")

            page = await s.query("h1")
            print(f"Logged in. Page title: {page.text}")


if __name__ == "__main__":
    asyncio.run(main())
